### PR TITLE
ci: migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by fisker**
Feel free to close if you don't want to do this.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
Comment (ikatyang) [inline .github/workflows/ci.yml:7]: Triggering CI from every commit and PR would be better I guess:

```suggestion
  - push
  - pull_request
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - actions/checkout@v2.3.4
+      - actions/setup-node@v2.1.2
+      - yarn install --frozen-lockfile
+      - yarn run lint
```
Comment (ikatyang) [inline .github/workflows/ci.yml:17]: Ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idsteps

I guess the `steps` are required to be a sequence of mapping:

```suggestion
      - uses: actions/checkout@v2.3.4
      - uses: actions/setup-node@v2.1.2
      - run: yarn install --frozen-lockfile
      - run: yarn run lint
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        node:
+          - "15"
+          - "14"
+          - "10"
```
Comment (ikatyang) [inline .github/workflows/ci.yml:26]: We should probably test our exact minimal supported version:

```suggestion
          - "10.3.0"
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      - actions/setup-node@v2.1.2
+      - yarn install --frozen-lockfile
+      - yarn run lint
+
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        node:
+          - "15"
```
Comment (ikatyang) [inline .github/workflows/ci.yml:24]: We probably do no need to test the non-LTS version:
```suggestion
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      - ENABLE_COVERAGE: 1
+    steps:
+      - actions/checkout@v2.3.4
+      - actions/setup-node@v2.1.2
+      - yarn install --frozen-lockfile
+      - yarn global add codecov
+      - if: matrix.angular != ''
+        run: yarn upgrade @angular/compiler@${{ matrix.angular }}
+      - yarn run lint
+      - codecov
```
Comment (ikatyang) [inline .github/workflows/ci.yml:43]: Ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idsteps

I guess the `steps` are required to be a sequence of mapping:

```suggestion
      - uses: actions/checkout@v2.3.4
      - uses: actions/setup-node@v2.1.2
        with: 
          node-version: ${{ matrix.node }}
      - run: yarn install --frozen-lockfile
      - run: yarn global add codecov
      - if: matrix.angular != ''
        run: yarn upgrade @angular/compiler@${{ matrix.angular }}
      - run: yarn run test
      - run: codecov
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+          - "14"
+          - "10"
+        angular:
+          - ""
+          - "^9.1.0"
+          - "~10.0.0"
+    name: Node.js ${{ matrix.node }} / angular ${{ matrix.angular }}
+    runs-on: ubuntu-latest
+    env:
+      - ENABLE_COVERAGE: 1
```
Comment (ikatyang) [inline .github/workflows/ci.yml:34]: Ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idenv

`env` should be a map:

```suggestion
      ENABLE_COVERAGE: 1
```

**Review by ikatyang — COMMENTED** [review]
Thanks for doing this! (Is there a way to trigger the GitHub Action for this PR?)

**Comment by fisker** [comment]
> Is there a way to trigger the GitHub Action for this PR?

I believe it can't run, untill master have a workflow with same name.

**Review by ikatyang — APPROVED** [review]

**Comment by ikatyang** [comment]
Thanks!

**Pull request merged**